agent/05_agent_create_cluster.sh

```diff
@@ -107,9 +107,9 @@ fi
 
 
 wait_for_cluster_ready
+
 # Temporary fix for the CI. To be removed once we'll 
 # be able to generate the cluster credentials
-if [ "${OPENSHIFT_CI}" == true ]; then
-  touch ${OCP_DIR}/auth/kubeadmin-password
+if [ ! -f "${OCP_DIR}/auth/kubeadmin-password" ]; then
+    oc patch --kubeconfig="${OCP_DIR}/auth/kubeconfig" secret -n kube-system kubeadmin --type json -p '[{"op": "replace", "path": "/data/kubeadmin", "value": "'"$(openssl rand -base64 18 | tr -d '\n' | tee "${OCP_DIR}/auth/kubeadmin-password" | htpasswd -nBi -C 10 "" | tr -d ':\n' | sed -e 's/\$2y\$/$2a$/' | base64 -w 0 -)"'"}]'
```

Comment:
```suggestion
set +x
    oc patch --kubeconfig="${OCP_DIR}/auth/kubeconfig" secret -n kube-system kubeadmin --type json -p '[{"op": "replace", "path": "/data/kubeadmin", "value": "'"$(openssl rand -base64 18 | tr -d '\n' | tee "${OCP_DIR}/auth/kubeadmin-password" | htpasswd -nBi -C 10 "" | tr -d ':\n' | sed -e 's/\$2y\$/$2a$/' | base64 -w 0 -)"'"}]'
set -x
```

Reply:
Why? The password is only ever written to a file, it never appears in the shell.

Reply:
From https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-metal3_dev-scripts/1439/pull-ci-openshift-metal3-dev-scripts-master-e2e-agent-compact/1560288088255631360/artifacts/e2e-agent-compact/baremetalds-devscripts-setup/build-log.txt - only the hash is printed:

```+(./agent/05_agent_create_cluster.sh:113): main(): '[' '!' -f ocp/ostest/auth/kubeadmin-password ']'
++(./agent/05_agent_create_cluster.sh:114): main(): openssl rand -base64 18
++(./agent/05_agent_create_cluster.sh:114): main(): tr -d '\n'
++(./agent/05_agent_create_cluster.sh:114): main(): tee ocp/ostest/auth/kubeadmin-password
++(./agent/05_agent_create_cluster.sh:114): main(): htpasswd -nBi -C 10 ''
++(./agent/05_agent_create_cluster.sh:114): main(): tr -d ':\n'
++(./agent/05_agent_create_cluster.sh:114): main(): sed -e 's/\$2y\$/$2a$/'
++(./agent/05_agent_create_cluster.sh:114): main(): base64 -w 0 -
+(./agent/05_agent_create_cluster.sh:114): main(): oc patch --kubeconfig=ocp/ostest/auth/kubeconfig secret -n kube-system kubeadmin --type json -p '[{"op": "replace", "path": "/data/kubeadmin", "value": "JDJhJDEwJHpWU0VhUEx6RXI1b0hZVzlhbzgzMU9VWjR3SGo1eGVONlFBTGh3aDJMY1Aua25ES1dEVy5t"}]'
secret/kubeadmin patched
```

Reply:
Isn't that just the base64 encoded value? If so any automatic tool parsing the logs could decode it, in such case it'd be better to remove it from the output

Reply:
It's the base-64 encoded value of the _hash_.

```
$ echo 'JDJhJDEwJHpWU0VhUEx6RXI1b0hZVzlhbzgzMU9VWjR3SGo1eGVONlFBTGh3aDJMY1Aua25ES1dEVy5t' | base64 -d
$2a$10$zVSEaPLzEr5oHYW9ao831OUZ4wHj5xeN6QALhwh2LcP.knDKWDW.m
```

Reply:
Sorry I missed the bcrypt part. I won't block on this, even though keeping the `+x` for critical sections like this one wouldn't hurt (ie, to avoid accidental issues in future PRs touching this zone)

Reply:
/approve